### ISSUE TITLE
Git clean deletes the content of symbolic links

### ISSUE THREAD

**Issue by lwinkler** [git-for-windows/git#2251]
- [X] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Git 64 bits on Windows 10
git version 2.20.1.windows.1

```
$ git --version --build-options

cpu: x86_64
built from commit: 7c9fbc07db0e2939b36095df45864b8cda19b64f
sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
  Windows 10 64 bits

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.18362.175]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

**defaults but I overrided the windows tools with bash tools**

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** Using cmder as console **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

** bash **

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
# How to reproduce
λ mkdir test-git
λ cd test-git/
λ touch a b c
λ cd ..
λ mkdir other_dir
λ cd other_dir/
λ touch d e f
λ cd ../test-git/
λ git init
λ git add .
λ git commit -am initial

# << create a symlink called other_dir to ../other_dir using windows command
mklink /J "other_dir" "..\other_dir"
 >>

λ ls other_dir/
d  e  f
λ git clean -fdx
Removing other_dir/
λ ls ../other_dir/
... nothing ... the directory is empty. This means that instead of removing the symbolic link as expected git clean entered the symlink and deleted its content. This is extremely dangerous behavior !

```

**Comment by rimrul** [comment]
This seems to be the same as #607.
The PR that should fix this is #1976. I don't know the current status of that PR, though...

**Comment by drizzd** [comment]
Closing as duplicate of #607.
